Source/Core/Core/Core.cpp

```diff
@@ -146,7 +163,16 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	#ifdef ThreadLocalStorage
+		return tls_is_cpu_thread;
+	#else
+		if(!s_has_init_cpu_tls)
```

Comment:
Your usage of s_has_init_cpu_tls here isn't threadsafe; s_has_init_cpu_tls needs to be a pthread_once_t or something like that.